lib/sdp/media.ex

```diff
@@ -159,4 +168,68 @@ defmodule Membrane.Protocol.SDP.Media do
   end
 
   defp gen_ports(port_no, _), do: [port_no]
+
+  defp header_fields(media) do
+    [
+      serialize_type(media.type),
+      serialize_ports(media.ports),
+      media.protocol,
+      serialize_fmt(media.fmt)
+    ]
+  end
+
+  defp other_fields(media) do
+    [
+      {"i", :title, nil},
+      {"c", :connection_data, Membrane.Protocol.SDP.ConnectionData},
+      {"b", :bandwidth, Membrane.Protocol.SDP.Bandwidth},
+      {"k", :encryption, Membrane.Protocol.SDP.Encryption},
+      {"a", :attributes, Membrane.Protocol.SDP.Attribute}
+    ]
+    |> Enum.map(&add_types(&1, media))
+    |> Enum.reject(&(&1 == ""))
+  end
+
+  def add_types({type_string, key, module}, media) do
+    case Map.get(media, key) do
+      nil ->
+        ""
+
+      list when is_list(list) ->
+        list
+        |> Enum.map(&serialize_optional(&1, module))
+        |> Enum.reject(&(&1 == ""))
+        |> Enum.map_join(&add_type(&1, type_string))
+
+      single_field ->
+        single_field |> serialize_optional(module) |> add_type(type_string)
+    end
+  end
+
+  defp add_type(string, type) do
+    if String.at(string, 1) == "=" do
```

Comment:
Is it possible that you will have a string including the type here? This if should not appear here IMO

Reply:
And since it's here, I advice to use keyword notation for one-liners. `if sth, do: sth, else: sth_else`